website/src/pages/[organism]/submission/[groupId]/index.astro

```diff
@@ -1,5 +1,4 @@
 ---
-import { cleanOrganism } from '../../../../components/Navigation/cleanOrganism';
```

Comment:
Why is the `cleanOrganism` function not needed anymore?

Reply:
It was never needed here and just complicated accessing the organism name, we can get it simply from the organism path.

Reply:
It doesn't seem to be doing anything useful, we don't use anything but organism!.key later on, so let's cut out the indirection

Reply:
If the user puts a garbage organism in the path, they'll get a 500 error. The fix seems to be to properly guard against this, like most other pages in the codebase do:

```ts
const organism = Astro.params.organism!;
const { organism: cleanedOrganism } = cleanOrganism(Astro.params.organism);

if (!cleanedOrganism) {
    return {
        statusCode: 404,
        body: 'Organism not found',
    };
}
```

Reply:
Maybe Astro middleware could be used to do this validation in a more DRY way.

https://docs.astro.build/en/guides/middleware

Reply:
I don't see a 500, where do you?
<img width="1216" alt="image" src="https://github.com/user-attachments/assets/70b9205b-ec58-4256-b29c-b54142f7e877">

Reply:
I didn't actually check, sorry. That was just my understanding from ChatGPT and the astro docs 😅 

In that case please don't do an incomplete change like this: remove _all_ usages of the function as well as the function itself!